runtime/binding-http/src/test/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpQueryStringComparatorTest.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.binding.http.internal.config;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+import org.junit.Test;
+
+public class HttpQueryStringComparatorTest
+{
+    @Test
+    public void shouldCompareSameUnencodedEqualsToUnencoded()
+    {
+        // GIVEN
+        HttpQueryStringComparator comparator = new HttpQueryStringComparator();
+
+        // WHEN
+        int result = comparator.compare("hello", "hello");
+
+        // THEN
+        assertThat(result, equalTo(0));
```

Comment:
Use [`comparator`](https://hamcrest.org/JavaHamcrest/javadoc/2.0.0.0/org/hamcrest/comparator/ComparatorMatcherBuilder.html#comparedBy-java.util.Comparator-) support in matchers.

```
assertThat("hello", comparedBy(comparator).comparesEqualTo("hello"));
```

Similar feedback for other test method implementations.